kernel/memory_structs/src/lib.rs

```diff
@@ -527,6 +666,20 @@ macro_rules! implement_page_frame_range {
                     self.0.iter()
                 }
             }
+            impl IntoIterator for $TypeName<Page2MiB> {
+                type Item = $chunk<Page2MiB>;
+                type IntoIter = RangeInclusiveIterator<$chunk<Page2MiB>>;
+                fn into_iter(self) -> Self::IntoIter {
+                    self.0.iter()
+                }
+            }
+            impl IntoIterator for $TypeName<Page1GiB> {
+                type Item = $chunk<Page1GiB>;
+                type IntoIter = RangeInclusiveIterator<$chunk<Page1GiB>>;
+                fn into_iter(self) -> Self::IntoIter {
+                    self.0.iter()
+                }
+            }
```

Comment:
why does this have to be done specifically for these two page sizes? you can just do it generically like with all other functions.

If you're not sure of the syntax, you can do this:
```rust
impl<P: PageSize> IntoIterator for $TypeName<P> {
    ...
}
```